src/sbt-test/docker/build-command/build.sbt

```diff
@@ -1,9 +1,17 @@
+import xsbti.FileConverter
+import com.typesafe.sbt.packager.PluginCompat
+import NativePackagerHelper._
+
 enablePlugins(JavaAppPackaging)
 
+scalaVersion := "2.12.20"
+
 name := "docker-build-command-test"
 
 version := "0.1.0"
 
-import NativePackagerHelper._
-Docker / mappings ++= directory("src/main/resources/docker-test")
```

Comment:
@eed3si9n should `sbt.io.Mapper#directory` and other methods in `sbt.io.Mapper` that return mappings (i.e `(File, String)`) be changed to return ( `(HashedVirtualFileRef, String)` ) so it aligns with the changes to `sbt.Keys#mappings` which changed from using `File` to  `HashedVirtualFileRef`?

Reply:
That makes sense.

Reply:
unfortunately `sbt.io.Mapper` is in [sbt/io](https://github.com/sbt/io/blob/develop/io/src/main/scala/sbt/io/PathMapper.scala#L141) which does not have access to `HashedVirtualFileRef`...

Reply:
WDYT about reverting `mappings` to `(File, String)`? I gave it a try [here](https://github.com/sbt/sbt/pull/7937)

Reply:
`File` simultaneously is specific to user's environment, and not specific enough about the content, so it's inappropriate to participate in remote cache. For anything useful, mappings is going to be a critical input, so we have to make this change if we want remote caching to be a major feature of sbt 2.x. more details here
https://eed3si9n.com/sbt-remote-cache-with-bazel-compat/

Reply:
> unfortunately sbt.io.Mapper is in [sbt/io](https://github.com/sbt/io/blob/develop/io/src/main/scala/sbt/io/PathMapper.scala#L141) which does not have access to HashedVirtualFileRef...

any tips on how to proceed here? remove the "mappings" related methods from `sbt.io.Mapper`, and put them somewhere else (where)?

Reply:
sbt/sbt for now?

Reply:
> sbt/sbt for now?

yes, but which package/class, so that is available in `build.sbt` files without import?

Reply:
We can put in almost any place that makes sense for `build.sbt`. The imports are https://github.com/sbt/sbt/blob/develop/main/src/main/scala/sbt/internal/BuildUtil.scala#L93-L100, and we control that with https://github.com/sbt/sbt/blob/develop/sbt-app/src/main/scala/sbt/Import.scala.

We can create `Mapper` in sbt package and put it in https://github.com/sbt/sbt/tree/develop/main-actions/src/main/scala/sbt for example, and remove the Mapper alias in https://github.com/sbt/sbt/blob/ebee3a86d900758b7085bb40979124ad2a9e4587/sbt-app/src/main/scala/sbt/Import.scala#L57

Reply:
Thanks. I will give it a try later today

Reply:
@eed3si9n I tried this in https://github.com/sbt/sbt/pull/7939 but I don't think this is enough. 

> File simultaneously is specific to user's environment, and not specific enough about the content, so it's inappropriate to participate in remote cache. For anything useful, mappings is going to be a critical input, so we have to make this change if we want remote caching to be a major feature of sbt 2.x. more details here
[eed3si9n.com/sbt-remote-cache-with-bazel-compat](https://eed3si9n.com/sbt-remote-cache-with-bazel-compat/)

I understand these concerns, but making user code so verbose will make adoption of sbt 2.x a pain as it is a downgrade in quality from sbt 1.x

> WDYT about reverting mappings to `(File, String)`? 

cant we keep mappings key as `(File, String)`, but move the cache implementation details to the task implementation? Even if it make task implementation more verbose?